feature/experimental/bgp/otg_tests/bgp_always_compare_med/bgp_always_compare_med_test.go

```diff
@@ -384,86 +399,104 @@ func verifyTraffic(t *testing.T, ate *ondatra.ATEDevice, c gosnappi.Config, flow
+	}
+}
+
+// configPolicy is used to configure routing policies on the DUT.
+func configPolicy(t *testing.T, dut *ondatra.DUTDevice, d *oc.Root) {
+
 	rp := d.GetOrCreateRoutingPolicy()
 
 	pdef1 := rp.GetOrCreatePolicyDefinition(setMEDPolicy100)
-	stmt1, err := pdef1.AppendNewStatement(aclStatement20)
+	st, err := pdef1.AppendNewStatement(aclStatement20)
 	if err != nil {
-		t.Errorf("Error while creating new statement %v", err)
+		t.Fatal(err)
 	}
-	actions1 := stmt1.GetOrCreateActions()
+	actions1 := st.GetOrCreateActions()
 	actions1.GetOrCreateBgpActions().SetMed = oc.UnionUint32(bgpMED100)
 
 	pdef2 := rp.GetOrCreatePolicyDefinition(setMEDPolicy50)
-	stmt2, err := pdef2.AppendNewStatement(aclStatement20)
+	st, err = pdef2.AppendNewStatement(aclStatement20)
 	if err != nil {
-		t.Errorf("Error while creating new statement %v", err)
+		t.Fatal(err)
 	}
-	actions2 := stmt2.GetOrCreateActions()
+	actions2 := st.GetOrCreateActions()
 	actions2.GetOrCreateBgpActions().SetMed = oc.UnionUint32(bgpMED50)
 
-	gnmi.Replace(t, dut, gnmi.OC().RoutingPolicy().Config(), rp)
-
-	// Apply setMed import policy on eBGP Peer1 - OTG Port2 - with MED 100.
-	dutPolicyConfPath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).
-		Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp().
-		PeerGroup(peerGrpName2).ApplyPolicy().ImportPolicy()
-	gnmi.Replace(t, dut, dutPolicyConfPath.Config(), []string{setMEDPolicy100})
+	pdef3 := rp.GetOrCreatePolicyDefinition(rplAllowPolicy)
+	st, err = pdef3.AppendNewStatement("id-1")
+	if err != nil {
+		t.Fatal(err)
+	}
+	action3 := st.GetOrCreateActions()
+	action3.PolicyResult = oc.RoutingPolicy_PolicyResultType_ACCEPT_ROUTE
 
-	// Apply setMed Import policy on eBGP Peer2 - OTG Port3 -  with MED 50.
-	dutPolicyConfPath = gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).
-		Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp().
-		PeerGroup(peerGrpName3).ApplyPolicy().ImportPolicy()
-	gnmi.Replace(t, dut, dutPolicyConfPath.Config(), []string{setMEDPolicy50})
+	gnmi.Replace(t, dut, gnmi.OC().RoutingPolicy().Config(), rp)
 }
 
 // verifySetMed is used to validate MED on received prefixes at OTG Port1.
-func verifySetMed(t *testing.T, otg *otg.OTG, config gosnappi.Config, wantMEDValue uint32) {
+func verifySetMed(t *testing.T, otg *otg.OTG, config gosnappi.Config, wantMEDValue uint32, expectedAggrRouteRxValue uint64) {
 	t.Helper()
-	_, ok := gnmi.WatchAll(t,
-		otg,
-		gnmi.OTG().BgpPeer(ateSrc.Name+".BGP4.peer").UnicastIpv4PrefixAny().State(),
-		2*time.Minute,
-		func(v *ygnmi.Value[*otgtelemetry.BgpPeer_UnicastIpv4Prefix]) bool {
-			return v.IsPresent()
+
+	// First verify if expected count of routes are recived by the BGP peer on test port1 and
+	// then verify if all routeCount prefixes in RIB/Learend Information have expected Med val
+	_, ok := gnmi.Watch(t, otg, gnmi.OTG().BgpPeer(ateSrc.Name+".BGP4.peer").Counters().InRoutes().State(),
```

Comment:
The OTG test isn't passing in general because of this check because we appear to be getting an count of 508 the first time we check this. (I don't know why that would be the case given we start the OTG protocols after we've configured the MED policy.) 

The ATE test does not have the equivalent of this check, so can we skip this and just check `gotPrefixCount == routeCount` below? (I think the 'verifyPrefixesTelemetry' function should suffice for validating that routes were sent.)

Reply:
Agreed, 'verifyPrefixesTelemetry' should suffice for validating the routes. Removed the check for routeCount